aquadoggo/src/network/replication/behaviour.rs

```diff
@@ -219,36 +287,63 @@ mod tests {
         let swarm1_peer_id = *swarm1.local_peer_id();
         let swarm2_peer_id = *swarm2.local_peer_id();
 
-        // Send a message from to swarm1 local peer from swarm2 local peer.
-        swarm1
-            .behaviour_mut()
-            .send_message(swarm2_peer_id, Message::Dummy(0));
+        // Send a message from swarm1 to peer2.
+        swarm1.behaviour_mut().send_message(
+            swarm2_peer_id,
+            ConnectionId::new_unchecked(0),
+            SyncMessage::new(0, Message::SyncRequest(0.into(), target_set_1.clone())),
+        );
 
-        // Send a message from to swarm2 local peer from swarm1 local peer.
-        swarm2
-            .behaviour_mut()
-            .send_message(swarm1_peer_id, Message::Dummy(1));
+        // Send a message from swarm2 peer1.
+        swarm2.behaviour_mut().send_message(
+            swarm1_peer_id,
+            ConnectionId::new_unchecked(0),
+            SyncMessage::new(1, Message::SyncRequest(0.into(), target_set_2.clone())),
+        );
 
         // Collect the next 2 behaviour events which occur in either swarms.
         for _ in 0..2 {
             tokio::select! {
-                BehaviourOutEvent::MessageReceived(peer_id, message) = swarm1.next_behaviour_event() => res1.push((peer_id, message)),
-                BehaviourOutEvent::MessageReceived(peer_id, message) = swarm2.next_behaviour_event() => res2.push((peer_id, message)),
+                Event::ConnectionEstablished(peer_id, _) = swarm1.next_behaviour_event() => res1.push((peer_id, None)),
+                Event::ConnectionEstablished(peer_id, _) = swarm2.next_behaviour_event() => res2.push((peer_id, None)),
+            }
+        }
+
+        // And again add the next 2 behaviour events which occur in either swarms.
+        for _ in 0..2 {
+            tokio::select! {
+                Event::MessageReceived(peer_id, message, _) = swarm1.next_behaviour_event() => res1.push((peer_id, Some(message))),
+                Event::MessageReceived(peer_id, message, _) = swarm2.next_behaviour_event() => res2.push((peer_id, Some(message))),
```

Comment:
This fails with "Peer id needs to be given" because of my change moving the `PeerId` to the `NetworkConfiguration` struct. I need to refactor this properly .. a quick change didn't do it.